scipy/stats/_continuous_distns.py

```diff
@@ -4648,487 +4647,6 @@ def _stats(self):
-            else:
-                warnings.warn('Density calculations experimental for FFT method.' +
-                              ' Use combination of zolatarev and quadrature methods instead.', RuntimeWarning)
-                _alpha, _beta = pair
-                _x = data_subset[:,(0,)]
-
-                # need enough points to "cover" _x for interpolation
-                h = fft_grid_spacing
-                q = np.ceil(np.log(2*np.max(np.abs(_x))/h)/np.log(2)) + 2 if fft_n_points_two_power is None else int(fft_n_points_two_power)
-
-                density_x, density = levy_stable_gen._pdf_from_cf_with_fft(lambda t: levy_stable_gen._cf(t, _alpha, _beta), h=h, q=q)
-                f = interpolate.interp1d(density_x, np.real(density))
-                data_out[data_mask] = f(_x)
-
-        return data_out.T[0]
-
-    def _cdf(self, x, alpha, beta):
-
-        x = np.asarray(x).reshape(1, -1)[0,:]
-
-        x, alpha, beta = np.broadcast_arrays(x, alpha, beta)
-
-        data_in = np.dstack((x, alpha, beta))[0]
-        data_out = np.empty(shape=(len(data_in),1))
-
-        fft_min_points_threshold = getattr(self, 'pdf_fft_min_points_threshold', None)
-        fft_grid_spacing = getattr(self, 'pdf_fft_grid_spacing', 0.001)
-        fft_n_points_two_power = getattr(self, 'pdf_fft_n_points_two_power', None)
-
-        # group data in unique arrays of alpha, beta pairs
-        uniq_param_pairs = np.vstack(
-            list({tuple(row) for row in data_in[:,1:]}))
-        for pair in uniq_param_pairs:
-            data_mask = np.all(data_in[:,1:] == pair, axis=-1)
-            data_subset = data_in[data_mask]
-            if fft_min_points_threshold is None or len(data_subset) < fft_min_points_threshold:
-                data_out[data_mask] = np.array([levy_stable._cdf_single_value_zolotarev(_x, _alpha, _beta)
-                            for _x, _alpha, _beta in data_subset]).reshape(len(data_subset), 1)
-            else:
-                warnings.warn("FFT method is considered experimental for "
-                              "cumulative distribution function "
-                              "evaluations. Use Zolotarev's method instead.",
-                              RuntimeWarning)
-                _alpha, _beta = pair
-                _x = data_subset[:,(0,)]
-
-                # need enough points to "cover" _x for interpolation
-                h = fft_grid_spacing
-                q = 16 if fft_n_points_two_power is None else int(fft_n_points_two_power)
-
-                density_x, density = levy_stable_gen._pdf_from_cf_with_fft(lambda t: levy_stable_gen._cf(t, _alpha, _beta), h=h, q=q)
-                f = interpolate.InterpolatedUnivariateSpline(density_x, np.real(density))
-                data_out[data_mask] = np.array([f.integral(self.a, x_1) for x_1 in _x]).reshape(data_out[data_mask].shape)
-
-        return data_out.T[0]
-
-    def _fitstart(self, data):
```

Comment:
Will this change remove the functionality of levy_stable._fitstart()?  I am working with data that has been aggregated to percentiles (3 to 97) and in some cases to quintiles (10, 25, 50, 75, 90).  The part of _fitstart() that uses quintiles to estimate stable parameters is useful for these data, especially if the code was re-factored a bit, but if _fitstart() is disappearing altogether then I guess I can either just embed the functionality within my analysis or propose new additions if it seems appropriate to put something like that in scipy.

Reply:
No, it's just moving it to a separate module. You can use it directly from here instead:

https://github.com/scipy/scipy/blob/113a699672b680de675bc7949b77ae4b7cc16b00/scipy/stats/_levy_stable_distn.py#L479-L481

Reply:
Better yet, moving `levy_stable_gen` to it's own file is only an internal implementation detail that doesn't change the public API. Any existing code that does something like:

```python
from scipy.stats import levy_stable

...

alpha, beta, delta, c = levy_stable._fitstart(data)
```

should continue to work.

Reply:
Yep, `_fitstart()` still exists and basically just passes the data to one of two functions handling the parameterization now.
https://github.com/scipy/scipy/blob/113a699672b680de675bc7949b77ae4b7cc16b00/scipy/stats/_levy_stable_distn.py#L1069-L1074

Note there is an ongoing PR fixing some small inconsistencies in `_fitstart()` w.r.t. McCulloch's paper (https://github.com/scipy/scipy/pull/14476). It's likely that gets merged before this PR, which will almost certainly give us some merge conflicts here, but luckily the changes are relatively minor.